src/main/java/org/nuxeo/ecm/core/storage/marklogic/MarkLogicHelper.java

```diff
@@ -58,11 +59,11 @@ public static Calendar deserializeCalendar(String calString) {
     }
 
     public static String serializeKey(String key) {
-        return key.replace(SCHEMA_ORIGINAL_DELIMITER, SCHEMA_MARKLOGIC_DELIMITER);
+        return SCHEMA_ORIGINAL_DELIMITER.matcher(key).replaceFirst(SCHEMA_MARKLOGIC_DELIMITER.pattern());
```

Comment:
If this is really an issue for speed or memory, you could write a simple character-based loop that does the replacement.